lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -2595,6 +2610,22 @@ def verify_session(session_id, quiet = false)
     end
   end
 
+  #
+  # Custom error code to handle timeout errors
+  #
+  # @param message [String] The message to be printed when a timeout error is hit
+
+  def log_on_timeout_error(message)
+    proc do |e|
+      handled = false
+      if e.is_a?(Rex::TimeoutError) || e.is_a?(Timeout::Error)
+        elog(e)
+        print_error(message)
+        :handled
+      end
```

Comment:
This might be cleaner as a guard clause

```suggestion
      next unless e.is_a?(Rex::TimeoutError) || e.is_a?(Timeout::Error)
      elog(e)
      print_error(message)
      :handled
```